testing/trino-testing-services/src/test/java/io/trino/testng/services/TestStaticInnerTestClassVerifier.java

```diff
@@ -0,0 +1,64 @@
+package io.trino.testng.services;
+
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.trino.testng.services.ReportStaticInnerTestClasses.performActionIfInnerClass;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestStaticInnerTestClassVerifier
+{
+    private static Optional<Class<?>> getEnclosingClass(Class<?> testClass)
+    {
+        return performActionIfInnerClass(testClass, enclosingClass -> {});
+    }
+
+    @Test
+    public void testValid()
+    {
+        assertThat(getEnclosingClass(TestStaticInnerTestClassVerifierValidClass.class))
+                .isEmpty();
+    }
+
+    @Test
+    public void testInvalid()
+    {
+        assertThat(getEnclosingClass(TestClassWithInnerTestClasses.TestInnerTestClass.class))
+                .isPresent()
+                .contains(TestClassWithInnerTestClasses.class);
+
+        assertThat(getEnclosingClass(TestClassWithMultipleInnerTestClasses.TestInnerTestClassOne.class))
+                .isPresent()
+                .contains(TestClassWithMultipleInnerTestClasses.class);
+
+        assertThat(getEnclosingClass(TestClassWithMultipleInnerTestClasses.TestInnerTestClassTwo.class))
+                .isPresent()
+                .contains(TestClassWithMultipleInnerTestClasses.class);
+    }
```

Comment:
The tested method is one-liner "is inner class". I don't think this test adds any value.

Reply:
I wanted to prove to myself the automation works, I can remove the `MultipleInnerTestClasses` cases but I think keeping at least the first testcase is helpful.

Reply:
if/once you apply https://github.com/trinodb/trino/pull/11316/files#r819513710, there is nothing this test actually tests.